contracts/FlowIDTableStaking.cdc

```diff
@@ -836,22 +838,58 @@ pub contract FlowIDTableStaking {
         /// based on the tokens that they have staked
         pub fun payRewards(_ rewardsBreakdownArray: [RewardsBreakdown]) {
 
+            let feeRewards: UFix64 = FlowFees.getFeeBalance()
+            var mintedRewards: UFix64 = 0.0
+
+            // Borrow the fee admin and withdraw all the fees that have been collected since the last rewards payment
+            let feeAdmin = FlowIDTableStaking.borrowFeesAdmin()
+            let feeVault <- feeAdmin.withdrawTokensFromFeeVault(amount: feeRewards)
+
             let allNodeIDs = FlowIDTableStaking.getNodeIDs()
 
             let flowTokenMinter = FlowIDTableStaking.account.borrow<&FlowToken.Minter>(from: /storage/flowTokenMinter)
                 ?? panic("Could not borrow minter reference")
 
             for rewardBreakdown in rewardsBreakdownArray {
                 let nodeRecord = FlowIDTableStaking.borrowNodeRecord(rewardBreakdown.nodeID)
-                nodeRecord.tokensRewarded.deposit(from: <-flowTokenMinter.mintTokens(amount: rewardBreakdown.nodeRewards))
-                emit RewardsPaid(nodeID: rewardBreakdown.nodeID, amount: rewardBreakdown.nodeRewards)
+                let nodeReward = rewardBreakdown.nodeRewards
+                if nodeReward <= feeVault.balance {
+                    nodeRecord.tokensRewarded.deposit(from: <-feeVault.withdraw(amount: nodeReward))
+                } else if feeVault.balance > 0.0 {
+                    let remainingReward = nodeReward - feeVault.balance
+                    let nodeRewardVault <- feeVault.withdraw(amount: feeVault.balance)
+                    nodeRewardVault.deposit(from: <-flowTokenMinter.mintTokens(amount: remainingReward))
+                    mintedRewards = mintedRewards + remainingReward
+                    nodeRecord.tokensRewarded.deposit(from: <-nodeRewardVault)
+                } else {
+                    nodeRecord.tokensRewarded.deposit(from: <-flowTokenMinter.mintTokens(amount: nodeReward))
+                    mintedRewards = mintedRewards + nodeReward
+                }
+                
+                emit RewardsPaid(nodeID: rewardBreakdown.nodeID, amount: nodeReward)
 
                 for delegator in rewardBreakdown.delegatorRewards.keys {
                     let delRecord = nodeRecord.borrowDelegatorRecord(delegator)
-                    delRecord.tokensRewarded.deposit(from: <-flowTokenMinter.mintTokens(amount: rewardBreakdown.delegatorRewards[delegator]!))
-                    emit DelegatorRewardsPaid(nodeID: rewardBreakdown.nodeID, delegatorID: delegator, amount: rewardBreakdown.delegatorRewards[delegator]!)
+                    let delegatorReward = rewardBreakdown.delegatorRewards[delegator]!
+                    if delegatorReward <= feeVault.balance {
```

Comment:
I don't think I really want to make it a separate function because it just makes it less clear what it is doing. I did simplify the logic in my latest commit though